update_checker.go

```diff
@@ -0,0 +1,44 @@
+//go:build !docker
+
+package main
+
+import (
+	"context"
+	"time"
+
+	"github.com/suborbital/subo/subo/release"
+	"github.com/suborbital/subo/subo/util"
+)
+
+const checkVersionTimeout = 500 * time.Millisecond
+
+func checkForUpdates() {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	versionChan := make(chan string)
+
+	go func() {
+		msg := ""
+		if version, err := release.CheckForLatestVersion(); err != nil {
+			msg = err.Error()
+		} else if version != "" {
+			msg = version
+		}
+
+		select {
+		case <-ctx.Done():
+		default:
+			versionChan <- msg
+		}
+	}()
+
+	util.LogInfo("hhelloo?")
+	select {
+	case msg := <-versionChan:
+		if msg != "" {
+			util.LogInfo(msg)
+		}
+	case <-time.After(checkVersionTimeout):
```

Comment:
I wonder if the `context.WithTimeout(context.Backgroud(), checkVersionTimeout)` could be used here. https://go.dev/play/p/_oG0Kgm7O19

Yes, technically the time.After checkversion timeout does the same thing, but since we already have a context, we might as well use it.